File: 231-power-of-two/power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n == 1)
            return true;
        if (n % 2 != 0 || n < 1)
            return false;
        while (n != 2){
            n = n / 2;
            if (n % 2 != 0)
                return false;
        }
        return true;
    }
};

File: 231-power-of-two/power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        # return bin(n).count("1") == 1 and n>0
        
        if n <= 0:
            return False
        while n > 1:
            if n % 2 == 0:
                n /= 2
            else:
                return False
        return True
        